jbpm-wb-process-runtime/jbpm-wb-process-runtime-client/src/main/java/org/jbpm/workbench/pr/client/editors/instance/log/RuntimeLogPresenter.java

```diff
@@ -53,55 +59,83 @@ public IsWidget getWidget() {
         return view;
     }
 
+    public void setCurrentProcessInstanceId(Long currentProcessInstanceId) {
+        this.currentProcessInstanceId = currentProcessInstanceId;
+    }
+
+    public String getCurrentProcessName() {
+        return currentProcessName;
+    }
+
+    public void setCurrentProcessName(String currentProcessName) {
+        this.currentProcessName = currentProcessName;
+    }
+
+    public void setCurrentServerTemplateId(String currentServerTemplateId) {
+        this.currentServerTemplateId = currentServerTemplateId;
+    }
+
+    public void setCurrentDeploymentId(String currentDeploymentId) {
+        this.currentDeploymentId = currentDeploymentId;
+    }
+
     public void refreshProcessInstanceData(final LogOrder logOrder,
                                            final LogType logType) {
+        processRuntimeDataService.call((List<RuntimeLogSummary> logs) -> {
+            final List<String> logsLine = new ArrayList<String>();
+            if (logOrder == LogOrder.ASC) {
+                Collections.reverse(logs);
+            }
+            logs.stream().forEach(rls -> addLogline(logsLine,
```

Comment:
Same here, addLogline could be changed into a mapper function to be used in .map in the logs stream. That way you can also easily test the String result for a single line.